### PR TITLE
Format and UserDoc: Add 'txt' format to default formats.

### LINKED ISSUE #474: Format: add '.txt' to default file formats for analysis
Format.java's list of DEFAULT_FORMAT_STRINGS does not contain '.txt'.

Since it is a common file extension, it should be added.

### PR DESCRIPTION
**Pull request by dlworldpeace**
Fixes #474 

Some of the files committed are of .txt extension. In order to be able to track these files, let's add this as one of the default format.

### CODE REVIEW

**Comment by eugenepeh** [comment]
Thanks for the contribution @dlworldpeace . 
Do help us to update the documentations too, thank you.

**Comment by dlworldpeace** [comment]
Ready for the next round of review. 

At the same time, waiting for #440 to be merged so that I
can remove the line in test case where I forced the list of
formats to be 'txt' only.

File: src/main/java/reposense/model/Format.java [hunk 1]
```diff
@@ -9,7 +9,8 @@
  */
 public class Format {
     public static final List<String> DEFAULT_FORMAT_STRINGS = Arrays.asList(
-            "adoc", "cs", "css", "fxml", "gradle", "html", "java", "js", "json", "jsp", "md", "py", "tag", "xml");
+            "adoc", "cs", "css", "fxml", "gradle", "html", "java", "js",
+            "json", "jsp", "md", "py", "tag", "xml", "txt");
```
Comment (yong24s) [inline src/main/java/reposense/model/Format.java]: Maintain the alphabetical order?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -264,7 +264,7 @@ Column Name | Explanation
 ----------- | -----------
 Repository's Location | The `GitHub URL` or `Disk Path` to the git repository e.g., `https://github.com/foo/bar.git` or `C:\Users\user\Desktop\GitHub\foo\bar`
 [Optional] Branch | The branch to analyze in the target repository e.g., `master`. Default: the default branch of the repo
-[Optional] File formats<sup>*</sup> | The file extensions to analyze. Default: `adoc;cs;css;fxml;gradle;html;java;js;json;jsp;md;py;tag;xml`
+[Optional] File formats<sup>*</sup> | The file extensions to analyze. Default: `adoc;cs;css;fxml;gradle;html;java;js;json;jsp;md;py;tag;xml;txt`
```
Comment (yong24s) [inline docs/UserGuide.md]: Same as above

File: docs/UserGuide.md [hunk 3]
```diff
@@ -226,7 +226,7 @@ In addition, there are some _optional_ extra parameters you can use to customize
   Example:`-since 21/10/2017`
 * **`-until END_DATE`**: The end date of analysis. The analysis excludes the end date. Format: `DD/MM/YYYY`<br>
   Example:`-until 21/10/2017`
-* **`-formats LIST_OF_FORMATS`**: A space-separated list of file extensions that should be included in the analysis. Default: `adoc cs css fxml gradle html java js json jsp md py tag xml`<br>
+* **`-formats LIST_OF_FORMATS`**: A space-separated list of file extensions that should be included in the analysis. Default: `adoc cs css fxml gradle html java js json jsp md py tag xml txt`<br>
```
Comment (yong24s) [inline docs/UserGuide.md]: Same as above

File: docs/UserGuide.md [hunk 4]
```diff
@@ -147,7 +147,7 @@ Note: all fields are optional unless specified otherwise.
 **Fields to provide _repository-level_ info**:
 
 * `ignoreGlobList`: Folders/files to ignore, specified using the [_glob format_](https://docs.oracle.com/javase/tutorial/essential/io/fileOps.html#glob).
-* `formats`: File formats to analyze. Default: `adoc cs css fxml gradle html java js json jsp md py tag xml`
+* `formats`: File formats to analyze. Default: `adoc cs css fxml gradle html java js json jsp md py tag xml txt`
```
Comment (yong24s) [inline docs/UserGuide.md]: Same as above

File: src/main/java/reposense/model/Format.java (hunk 1 above)
Comment (dlworldpeace) [reply src/main/java/reposense/model/Format.java]: Tks for pointing it out.

**Comment by dlworldpeace** [comment]
Ready for next round of  review.

**Comment by dlworldpeace** [comment]
> LGTM

tks. but can wait for #440 to be merged first so that I
can remove the line in test case where I forced the list of
formats to be 'txt' only.

**Comment by yamidark** [comment]
>but can wait for #440 to be merged first so that I
can remove the line in test case where I forced the list of
formats to be 'txt' only.

Do you mean this should be merged 1st, so you can remove that line in #440 instead?

**Comment by dlworldpeace** [comment]
> > but can wait for #440 to be merged first so that I
> > can remove the line in test case where I forced the list of
> > formats to be 'txt' only.
> 
> Do you mean this should be merged 1st, so you can remove that line in #440 instead?

yes

**Comment by yamidark** [comment]
Nits to proposed merge message:
```
Format#DEFAULT_FORMAT_STRINGS contains a list of common file extensions
within git repositories, which RepoSense will use to run the analysis by default
if no formats is given by the user.

However, this list does not contain the `txt` file extension, which is
also another commonly used file extension in git repositories.

For user's convenience, let's add the `txt` file extension to the list
of default format strings as well.
```

**Comment by yamidark** [comment]
@eugenepeh Can help review my proposed message as well?

**Comment by eugenepeh** [comment]
LGTM

> At the same time, waiting for #440 to be merged so that I
can remove the line in test case where I forced the list of
formats to be 'txt' only.

Merging this one first so that there will be lesser unnecessary changes. 
The removal should be done in #440 itself.

**Pull request merged**